range.hpp

```diff
@@ -166,6 +165,12 @@ class iter::impl::Range {
     }
 
    public:
+    using iterator_category = std::forward_iterator_tag;
+    using value_type = T;
+    using difference_type = std::ptrdiff_t;
+    using pointer = value_type*;
+    using reference = value_type&;
```

Comment:
here's the annoying thing, the way range() is set up, reference is still `T` rather than `T&`. I've stressed over this a lot and [questioned std::vector<bool>::iterator in the process](https://stackoverflow.com/questions/30417500/requirements-for-forward-iterator-iterator-traitsreference), unfortunately reference = T is the current state of things